### PR TITLE
New Strategy: Implementation of Adaptor

### PR DESCRIPTION
**Pull request by marcharper**
Implements two strategies from [this paper](https://pdfs.semanticscholar.org/4656/7ef01c0554ce903595005f605f9f6bea6808.pdf)

* I wasn't able to reproducible Table 5 in the paper (see the tests) but otherwise it looks ok to me.
* Both strategies were named Adaptor in the paper
* ~This PR is based on top of #1212~

### CODE REVIEW

**Comment by drvinceknight** [comment]
@marcharper not entirely sure how/why it's failing the unit test (something stochastic) and also looks like coverage is missing line 40 of `adaptor.py`. (Could you also rebase on to master as #1212 is now in.)

**Comment by marcharper** [comment]
Rebased to before prompt-toolkit changes, we can rebase onto something else before merge.

File: axelrod/game.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def RPST(self) -> Tuple[Score, Score, Score, Score]:
         P = self.scores[(D, D)][0]
         S = self.scores[(C, D)][0]
         T = self.scores[(D, C)][0]
-        return (R, P, S, T)
+        return R, P, S, T
```
[the remark this answers is not in the document]
Comment (drvinceknight) [reply axelrod/game.py]: I believe this is esthetic and I remember reading or seeing a talk that the change is (for reasons unknown to me) preferred (fyi both styles are left unchanged by `black`). I believe it still fits the typing of the signature (type is unchanged):

```
>>> def main(a, b): return a, b                                                                                                                                 
>>> type(main(5, 3)) is tuple
True
```

[the remark this answers is not in the document]
Comment (meatballs) [reply axelrod/game.py]: Well, this project has always been an education and here's another example! Thanks

[the remark this answers is not in the document]
Comment (drvinceknight) [reply axelrod/game.py]: Tell me about it! "The Axelrod project" might as well be an item on my CV in the Education section. :laughing:

File: axelrod/strategies/adaptor.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        "stochastic": True,
+        "makes_use_of": set(),
+        "long_run_time": False,
+        "inspects_source": False,
+        "manipulates_source": False,
+        "manipulates_state": False,
+    }
+
+    def __init__(self, d: Dict[Tuple[Action, Action], float],
+                 perr: float = 0.01) -> None:
```
[the remark this answers is not in the document]
Comment (marcharper) [reply axelrod/strategies/adaptor.py]: Done. `perr` is the name in the paper and `d` --> `delta`

File: axelrod/game.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcharper) [reply axelrod/game.py]: I suspect it's slightly more efficient not to wrap the tuple in this case.

File: axelrod/strategies/adaptor.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+
+class AbstractAdaptor(Player):
+    """
+    An adaptive strategy that updates an internal state based on the last
+    round of play. Using this state the player Cooperates with a probability
+    derived from the state.
+
+    Parameters
+    ----------
```
Comment (drvinceknight) [inline axelrod/strategies/adaptor.py]: The `----...` need to be omitted for strategy docstrings (our setup can't handle them).

**Pull request merged**